sympy/parsing/c/c_parser.py

```diff
@@ -234,16 +236,43 @@ def transform_var_decl(self, node):
                 while child.kind == cin.CursorKind.NAMESPACE_REF:
                     child = next(children)
 
-                while child.kind == cin.CursorKind.TYPE_REF:
-                    child = next(children)
+                # while child.kind == cin.CursorKind.TYPE_REF:
+                #     child = next(children)
 
                 val = self.transform(child)
                 # List in case of variable assignment,
                 # FunctionCall node in case of a function call
                 if (child.kind == cin.CursorKind.INTEGER_LITERAL
+                    or child.kind == cin.CursorKind.FLOATING_LITERAL
                     or child.kind == cin.CursorKind.UNEXPOSED_EXPR):
-                    if (node.type.kind == cin.TypeKind.INT):
-                        type = IntBaseType(String('integer'))
+
+                    supported_int_types = [cin.TypeKind.SCHAR,
+                    cin.TypeKind.SHORT, cin.TypeKind.INT,
+                    cin.TypeKind.LONG, cin.TypeKind.UCHAR,
+                    cin.TypeKind.USHORT, cin.TypeKind.UINT,
+                    cin.TypeKind.ULONG]
+
+                    supported_float_types = [cin.TypeKind.FLOAT,
+                    cin.TypeKind.DOUBLE, cin.TypeKind.LONGDOUBLE]
+
+                    if node.type.kind in supported_int_types:
+                        if node.type.kind == cin.TypeKind.SCHAR:
+                            type = int8
+                        elif node.type.kind == cin.TypeKind.SHORT:
+                            type = int16
+                        elif node.type.kind == cin.TypeKind.INT:
+                            type = IntBaseType(String('integer'))
```

Comment:
you can use `intc` for this

Reply:
Yes, that would be better as it replicates `int` of C, working on it!